exchange/utils.go

```diff
@@ -313,105 +313,3 @@ func randomizeList(list []openrtb_ext.BidderName) {
 		list[i], list[j] = list[j], list[i]
 	}
 }
-
-func applyRegs(bidRequest *openrtb.BidRequest, isAMP, applyGDPR, applyCOPPA bool) {
-	if bidRequest.User != nil {
-		// Need to duplicate pointer objects
-		user := *bidRequest.User
-		bidRequest.User = &user
-
-		// There's no way for AMP to send a GDPR consent string yet so it's hard
-		// to know if the vendor is consented or not and therefore for AMP requests
-		// we keep the BuyerUID as is
-		if !isAMP {
-			bidRequest.User.BuyerUID = ""
-		}
-		if applyCOPPA {
-			bidRequest.User.ID = ""
-			bidRequest.User.Yob = 0
-			bidRequest.User.Gender = ""
-			bidRequest.User.BuyerUID = ""
-		}
-		bidRequest.User.Geo = cleanGeo(bidRequest.User.Geo, applyGDPR, applyCOPPA)
-	}
-	if bidRequest.Device != nil {
-		// Need to duplicate pointer objects
-		device := *bidRequest.Device
-		bidRequest.Device = &device
-
-		bidRequest.Device.DIDMD5 = ""
-		bidRequest.Device.DIDSHA1 = ""
-		bidRequest.Device.DPIDMD5 = ""
-		bidRequest.Device.DPIDSHA1 = ""
-		if applyCOPPA {
-			bidRequest.Device.MACSHA1 = ""
-			bidRequest.Device.MACMD5 = ""
-			bidRequest.Device.IFA = ""
-		}
-		bidRequest.Device.IP = cleanIP(bidRequest.Device.IP)
-		bidRequest.Device.IPv6 = cleanIPV6(bidRequest.Device.IPv6, applyGDPR, applyCOPPA)
-		bidRequest.Device.Geo = cleanGeo(bidRequest.Device.Geo, applyGDPR, applyCOPPA)
-	}
-}
-
-// Zero the last byte of an IP address
-func cleanIP(fullIP string) string {
-	i := strings.LastIndex(fullIP, ".")
-	if i == -1 {
-		return ""
-	}
-	return fullIP[0:i] + ".0"
-}
-
-func cleanIPV6(fullIP string, applyGDPR, applyCOPPA bool) string {
-	// If neither GDPR nor COPPA applies then do nothing
-	if !applyGDPR && !applyCOPPA {
-		return fullIP
-	}
-
-	i := strings.LastIndex(fullIP, ":")
-	if i == -1 {
-		return ""
-	}
-	fullIP = fullIP[0:i]
-
-	// If COPPA then remove the lowest 32 bits of the IP
-	if applyCOPPA {
-		fullIP = fullIP[:strings.LastIndex(fullIP, ":")+1] + "0:0"
-		return fullIP
-	}
-	// If GDPR then remove the lowest 32 bits of the IP
-	if applyGDPR {
-		fullIP = fullIP + ":0"
-	}
-	return fullIP
-}
-
-// Return a cleaned Geo object pointer (round off the latitude/longitude)
-func cleanGeo(geo *openrtb.Geo, applyGDPR, applyCOPPA bool) *openrtb.Geo {
-	// If neither GDPR nor COPPA applies then do nothing
-	if !applyCOPPA && !applyGDPR {
-		return geo
-	}
-
-	if geo == nil {
-		return nil
-	}
-	newGeo := *geo
-
-	// If GDPR applies then round off the Lat and LON values
-	if applyGDPR {
-		newGeo.Lat = float64(int(geo.Lat*100.0+0.5)) / 100.0
-		newGeo.Lon = float64(int(geo.Lon*100.0+0.5)) / 100.0
-	}
-
-	// If COPPA applies then remove Lat, Lon, Metro, City and Zip values
-	if applyCOPPA {
-		newGeo.Lat = 0
-		newGeo.Lon = 0
-		newGeo.Metro = ""
-		newGeo.City = ""
-		newGeo.ZIP = ""
-	}
-	return &newGeo
-}
```

Comment:
Good job consolidating all this functionality in the scrubber.go file. Looks neat

Reply:
Thnx. :)